src/github_permission_manager_webhook/tests/test_handler_github_permission_manager.py

```diff
@@ -0,0 +1,244 @@
+
+        mock_post.assert_called_once_with(
+            'https://api.github.com/repos/org/repo/issues/1/comments',
+            headers=github_permission_manager.auth_headers,
+            json={"body": comment_body}
+        )
+
+        assert mock_post.called
+
+def test_make_owner_on_github(github_permission_manager):
+    payload = {
+        'organization': {'login': 'test-org'}
+    }
+    user = 'test-user'
+
+    with patch('requests.put') as mock_put:
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_put.return_value = mock_response
+
+        github_permission_manager.make_owner_on_github(payload, user)
+
+        mock_put.assert_called_once_with(
+            'https://api.github.com/orgs/test-org/memberships/test-user',
+            headers=github_permission_manager.auth_headers,
+            json={"role": "admin"}
+        )
+
+        assert mock_response.status_code == 200
+
+def test_is_team_member(github_permission_manager):
+    user = 'user'
+    org = 'org'
+    team = 'team'
+
+    with patch('requests.get') as mock_get:
+        mock_response_team = MagicMock()
+        mock_response_team.json.return_value = {'id': 1}
+        mock_response_team.status_code = 200
+
+        mock_response_membership = MagicMock()
+        mock_response_membership.status_code = 200
+
+        mock_get.side_effect = [mock_response_team, mock_response_membership]
+
+        result = github_permission_manager.is_team_member(user, org, team)
+
+        mock_get.assert_any_call(
+            f'https://api.github.com/orgs/{org}/teams/{team}',
+            headers=github_permission_manager.auth_headers
+        )
+
+        mock_get.assert_any_call(
+            f'https://api.github.com/teams/1/memberships/{user}',
+            headers=github_permission_manager.auth_headers
+        )
+
+        assert result == True
+
+def test_handle_issue(github_permission_manager):
+    payload = {
+        'action': 'opened',
+        'issue': {'title': 'Request elevation', 'user': {'login': 'user'}},
+        'repository': {'owner': {'login': 'org'}}
+    }
+    with patch.object(github_permission_manager, '_is_user_eligible_for_elevation', return_value=True), \
+        patch.object(github_permission_manager, '_process_elevation_request') as mock_process:
+        github_permission_manager.handle_issue(json.dumps(payload))
+        mock_process.assert_called_once()
+
+def test_handle_issue_comment(github_permission_manager):
+    payload = {
+        'action': 'created',
+        'comment': {'user': {'login': 'user'}, 'body': 'approve'},
+        'repository': {'owner': {'login': 'org'}}
+    }
+    with patch.object(github_permission_manager, '_is_user_eligible_for_elevation', return_value=True), \
+        patch.object(github_permission_manager, '_handle_approval_comment') as mock_handle:
+        github_permission_manager.handle_issue_comment(json.dumps(payload))
+        mock_handle.assert_called_once()
```

Comment:
I'm a bit confused what these two tests are doing, it's mocking two functions and then asserting that one of them is called?

Reply:
Its testing the flow through the handle_issue() function, mocking the return value as true for the _is_user_eligible_for_elevation() call. I've added a test for ineligible users as well, where the return value is false.